libs/sdk-py/langgraph_sdk/encryption/__init__.py

```diff
@@ -0,0 +1,454 @@
+"""Custom encryption support for LangGraph.
+
+This module provides a framework for implementing custom at-rest encryption
+in LangGraph applications. Similar to the Auth system, it allows developers
+to define custom encryption and decryption handlers that are executed
+server-side.
+"""
+
+from __future__ import annotations
+
+import typing
+
+from langgraph_sdk.encryption import types
+
+
+class _JsonEncryptDecorators:
+    """Dynamic decorator factory for JSON encryption handlers.
+
+    Supports both default and model-specific handlers:
+    - @encrypt.json - default handler for all models
+    - @encrypt.json.thread - handler for thread model
+    - @encrypt.json("custom") - handler for custom model name
+    """
+
+    def __init__(self, parent: Encrypt):
+        self._parent = parent
+
+    def __call__(
+        self, fn_or_model: types.JsonEncryptor | str
+    ) -> (
+        types.JsonEncryptor
+        | typing.Callable[[types.JsonEncryptor], types.JsonEncryptor]
+    ):
+        """Register a JSON encryption handler.
+
+        Can be used as:
+        - @encrypt.json - register default handler
+        - @encrypt.json("model") - register model-specific handler
+
+        Args:
+            fn_or_model: Either the handler function (default) or model name string
+
+        Returns:
+            The registered handler function, or a decorator if model name provided
+        """
+        if isinstance(fn_or_model, str):
+            model_name = fn_or_model
+
+            def decorator(fn: types.JsonEncryptor) -> types.JsonEncryptor:
+                self._parent._json_encryptors[model_name] = fn
+                return fn
+
+            return decorator
+        else:
+            fn = fn_or_model
+            self._parent._json_encryptor = fn
+            return fn
+
+    def __getattr__(
```

Comment:
ooc: any reason why we shouldn't explicitly add assistants, threads, etc. so you get better type hinting/checking?